### PR TITLE
Get Searchable fields (task #1615)

### PR DESCRIPTION
**Pull request by flangofas**

### CODE REVIEW

File: src/Controller/Component/SearchableComponent.php [hunk 1]
```diff
@@ -16,4 +20,76 @@ class SearchableComponent extends Component
...
+
+        if (empty($path)) {
+            throw new RuntimeException('Please provide path of Tables to proceed.');
+        }
+
+        if (!file_exists($path)) {
+            return $result;
+        }
+
+        $it = new FilesystemIterator($path, FilesystemIterator::SKIP_DOTS);
```
Comment (mamchenkov) [inline src/Controller/Component/SearchableComponent.php]: @flangofas there must be a way to get all tables using Cake functionality ...

Comment (flangofas) [inline src/Controller/Component/SearchableComponent.php]: @mamchenkov Yeap, there is using the [Schema Collections](http://book.cakephp.org/3.0/en/orm/schema-system.html#schema-collections) but you cannot load the model without knowing the plugin. Like here: https://github.com/QoboLtd/cakephp-search/pull/1/files#diff-64b8a201ce5d538a2c857c90eebfe483R37

File: src/Controller/Component/SearchableComponent.php [hunk 2]
```diff
@@ -16,4 +17,77 @@ class SearchableComponent extends Component
...
+                    $this->loadModel($container . '.' . $table);
+                }
+                if (!method_exists($this->{$table}, 'getSearchableFields')) {
+                    unset($tables[$container][$key]);
+                }
+            }
+        }
+
+        return array_filter($tables);
+    }
```
Comment (georgeconstantinou) [inline src/Controller/Component/SearchableComponent.php]: @flangofas for this we will also need another similar wrapper method (probably you will need to rename this one) that will check if the table is searchable by checking if such method exists and call it. An example of the `isSearchable()` method can be found [here](https://github.com/QoboLtd/cakephp-csv-migrations/blob/master/src/ConfigurationTrait.php#L63-L70)

Comment (flangofas) [inline src/Controller/Component/SearchableComponent.php]: @georgeconstantinou Yeap. WIll be done next PR.

**Pull request merged**